rpmlint/checks/FileDigestCheck.py

```diff
@@ -90,47 +90,60 @@ def _is_valid_digest(self, pkgfile, digest, pkg):
+        """ Check all secured files of a group type
 
+        Ensures that all files in secured paths have to be whitelisted in an
+        digest whitelisting belonging to group_type.
+
+        Params:
+        - group_type: type of digest group type e.g. "cron", "dbus", ...
+        - secured_paths: all secured paths found in this package e.g. ['/usr/share/dbus-1/system-services/org.freedesktop.PolicyKit1.service', '/usr/share/dbus-1/system.d/org.freedesktop.PolicyKit1.conf']
+        """
+        # Find all digest whitelisted paths that belong to group_type and focus on current package
+        digests = []
         for digest_group in self.digest_groups:
-            if digest_group['type'] == group_type:
-                errors = self._calculate_errors_for_digest_group(pkg, digest_group, secured_paths)
-                if not errors:
-                    return []
-                if not best_errors or len(errors) < len(best_errors):
-                    best_errors = errors
-        return best_errors
+            if digest_group['type'] == group_type and digest_group['package'] == pkg.name:
+                digests.extend(digest_group['digests'])
+
+        # For all files in this package that fall into the secured paths: check if they are whitelisted
+        # If not whitelisted print error: file-digest-unauthorized
+        whitelisted_paths = {dg['path'] for dg in digests}
+        unauthorized = secured_paths - whitelisted_paths
+        for filename in unauthorized:
+            self.output.add_info('E', pkg, f'{group_type}-file-digest-unauthorized', filename, None)
+
+        # For all digest whitelisted files check if the digests in the package are correct
+        # If not correct print error: file-digest-mismatch
+        for path in whitelisted_paths:
+            # Find all digests with same path
+            # This is needed because there could be an older and a newer
+            # version of this package with same whitelisted paths and different digests
+            digests_of_path = []
+            for digest in digests:
+                if digest['path'] == path:
+                    digests_of_path.append(digest)
+            # If *any* digest with the same path matches the package's file
+            # digest of that path, then we assume the file is correctly whitelisted
+            error = True
+            error_digests = []
+            for digest in digests_of_path:
+                filename = digest['path']
+                # Check if digest whitelist path has a matching file in our package
+                if not pkg.files.get(filename):
+                    # This digest entry is not needed anymore and could be dropped
+                    continue
+                valid_digest, file_digest = self._is_valid_digest(pkg.files[filename], digest, pkg)
+                if valid_digest:
+                    # Valid digest found, no mismatch error will be printed
+                    error = False
```

Comment:
to make the code a bit more compact you could consider dropping the `error` variable and simply reset `error_digests = []`. Or do you think it hurts readability of the code?